Rubberduck.VBEEditor/ComManagement/ProjectsRepository.cs

```diff
@@ -35,13 +42,53 @@ private void LoadProjects()
             {
                 if (project.Protection == ProjectProtection.Locked)
                 {
-                    project.Dispose();
-                    continue;
+                    if (!TryStoreLockedProject(project))
+                    {
+                        project.Dispose();
+                    }
+                }
+                else
+                {
+                    EnsureValidProjectId(project);
+                    _projects.Add(project.ProjectId, project);
                 }
+            }
+        }
+
+        private bool TryStoreLockedProject(IVBProject project)
+        {
+            if (!TryGetFullPath(project, out var path))
+            {
+                _logger.Warn("Path of locked project could not be read.");
+            }
+
+            var projectName = project.Name;
+            var projectId = QualifiedModuleName.GetProjectId(projectName, path);
+
+            _lockedProjects.Add(projectId, project);
+            return true;
+        }
 
-                EnsureValidProjectId(project);
-                _projects.Add(project.ProjectId, project);
+        private bool TryGetFullPath(IVBProject project, out string fullPath)
```

Comment:
I've seen this method before (in the UserProjectRepository). Seems like that should be an extension onto `IVBProject` itself

Reply:
Although I am usually not huge fan of extension methods, it makes sense to me use one here.